x-pack/plugins/apm/public/components/app/correlations/ml_latency_correlations.tsx

```diff
@@ -0,0 +1,428 @@
+        <EuiFlexItem grow={false}>
+          {!isRunning && (
+            <EuiButton size="s" onClick={startFetch}>
+              <FormattedMessage
+                id="xpack.apm.correlations.latencyCorrelations.refreshButtonTitle"
+                defaultMessage="Refresh"
+              />
+            </EuiButton>
+          )}
+          {isRunning && (
+            <EuiButton size="s" onClick={cancelFetch}>
+              <FormattedMessage
+                id="xpack.apm.correlations.latencyCorrelations.cancelButtonTitle"
+                defaultMessage="Cancel"
+              />
+            </EuiButton>
+          )}
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiFlexGroup direction="column" gutterSize="none">
+            <EuiFlexItem>
+              <EuiText size="xs" color="subdued">
+                <FormattedMessage
+                  id="xpack.apm.correlations.latencyCorrelations.progressTitle"
+                  defaultMessage="Progress: {progress}%"
+                  values={{ progress: Math.round(progress * 100) }}
+                />
+              </EuiText>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiProgress
+                aria-label={i18n.translate(
+                  'xpack.apm.correlations.latencyCorrelations.progressAriaLabel',
+                  { defaultMessage: 'Progress' }
+                )}
+                value={Math.round(progress * 100)}
+                max={100}
+                size="m"
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      <EuiSpacer size="m" />
+      {overallHistogram !== undefined ? (
+        <>
+          <EuiTitle size="xxs">
+            <h4>
+              {i18n.translate(
+                'xpack.apm.correlations.latencyCorrelations.chartTitle',
+                {
+                  defaultMessage: 'Latency distribution for {name}',
+                  values: {
+                    name:
+                      fetchOptions.transactionName ?? fetchOptions.serviceName,
+                  },
+                }
+              )}
+            </h4>
+          </EuiTitle>
+
+          <CorrelationsChart
+            markerPercentile={DEFAULT_PERCENTILE_THRESHOLD}
+            markerValue={percentileThresholdValue ?? 0}
+            {...selectedHistogram}
+            overallHistogram={overallHistogram}
+          />
+
+          <EuiSpacer size="s" />
+        </>
+      ) : null}
+
+      {histograms.length > 0 && selectedHistogram !== undefined && (
+        <CorrelationsTable
+          // @ts-ignore correlations don't have the same column format other tables have
+          columns={mlCorrelationColumns}
+          // @ts-expect-error correlations don't have the same significant term other tables have
+          significantTerms={histogramTerms}
+          status={FETCH_STATUS.SUCCESS}
+          setSelectedSignificantTerm={setSelectedSignificantTerm}
```

Comment:
why passing a setter instead of actual significant terms? it's more difficult to follow and doesn't seem like a common practice.